CromIAM/src/main/scala/cromiam/webservice/RequestSupport.scala

```diff
@@ -32,4 +39,21 @@ trait RequestSupport {
       request <- extractStrictRequest
     } yield (user, request)
   }
+
+  def forwardIfUserEnabled(user: User, req: HttpRequest, cromwellClient: CromwellClient, samClient: SamClient): Route = {
+    import cromwell.api.model.EnhancedFailureResponseOrHttpResponseT
+
+    onComplete(samClient.isUserEnabledSam(user, req).value.unsafeToFuture()) {
+      case Success(Left(httpResponse: HttpResponse)) => complete(httpResponse)
+      case Success(Right(isEnabled: Boolean)) =>
+        authorize(isEnabled) {
+          complete {
+            cromwellClient.forwardToCromwell(req).asHttpResponse
+          }
+        }
+      case Failure(e) =>
+        val message = s"Unable to look up enablement status for user ${user.userId}: ${e.getMessage}"
```

Comment:
Should we also add "Please try again in sometime" to this error message so that the user knows that it was a system error and it might work if they retry?